libethereum/CodeSizeCache.h

```diff
@@ -0,0 +1,79 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file CodeSizeCache.h
+ * @date 2016
+ */
+
+#pragma once
+
+#include <map>
+#include <libdevcore/FixedHash.h>
+#include <libdevcore/Guards.h>
+
+namespace dev
+{
+namespace eth
+{
+
+/**
+ * @brief Simple thread-safe cache to store a mapping from code hash to code size.
+ * If the cache is full, a random element is removed.
+ */
+class CodeSizeCache
+{
+public:
+	void store(h256 const& _hash, size_t size)
+	{
+		UniqueGuard g(x_cache);
+		if (m_cache.size() > c_maxSize)
```

Comment:
Not related to the test failure, but `>=` would fit better with the name `c_maxSize`.  Currently the size can grow to `c_maxSize + 1`.